app/actors/sufia/interpret_visibility_actor.rb

```diff
@@ -0,0 +1,107 @@
+      end
+
+      # Validate that a lease is allowed by AdminSet's PermissionTemplate
+      def validate_lease(template)
+        # Leases are only allowable if a template doesn't require a release period or have any specific visibility requirement
+        # (Note: permission template release/visibility options do not support leases)
+        return true unless @intention.wants_lease? && template.present? && (template.release_period.present? || template.visibility.present?)
+
+        curation_concern.errors.add(:visibility, 'Lease option is not allowed by permission template for selected AdminSet.')
+        false
+      end
+
+      # Validate the selected release settings against template, checking for when embargoes/leases are not allowed
+      def validate_release_type(template)
+        # It's valid as long as embargo is not specified when a template requires no release delays
+        return true unless @intention.wants_embargo? && template.present? && template.release_no_delay?
+
+        curation_concern.errors.add(:visibility, 'Visibility specified does not match permission template "no release delay" requirement for selected AdminSet.')
+        false
+      end
+
+      # Validate visibility complies with AdminSet template requirements
+      def validate_visibility(attributes, template)
+        # NOTE: For embargo/lease, attributes[:visibility] will be nil (see sanitize_params), so visibility will be validated as part of embargo/lease
+        return true unless template.present? && attributes[:visibility].present?
+
+        # Validate against template's visibility requirements
+        return true if template.valid_visibility?(attributes[:visibility])
+
+        curation_concern.errors.add(:visibility, 'Visibility specified does not match permission template visibility requirement for selected AdminSet.')
+        false
+      end
+
+      # When specified, validate embargo is a future date that complies with AdminSet template requirements (if any)
+      def validate_embargo(attributes, template)
+        return true unless @intention.wants_embargo?
+
+        embargo_date = parse_date(attributes[:embargo_release_date])
+
+        # When embargo required, date must be in future AND matches any template requirements
+        return true if valid_future_date?(embargo_date) && valid_template_embargo_date?(embargo_date, template) && valid_template_visibility_after_embargo?(attributes, template)
+
+        curation_concern.errors.add(:visibility, 'When setting visibility to "embargo" you must also specify embargo release date.') unless embargo_date.present?
+        false
+      end
+
+      # Validate a date string is both a date AND in the future
+      def valid_future_date?(date)
+        return true if date.present? && date.future?
+
+        curation_concern.errors.add(:embargo_release_date, "Must be a future date.")
```

Comment:
The method name indicates that you are validating a future_date. However, you are adding the error to the embargo_release_date. There is an assumption of the scope of this method.

```ruby
def valid_future_date?(date, attribute_name: :embargo_release_date)
  return true if date.present?
  curation_concern.errors.add(attribute_name, "Must be a future date.")
end
```

Though to remove the embedded assumption, I encourage you to then change `valid_future_date?(embargo_data, attribute_name: :embargo_release_date)`